src/OrchardCore.Modules/OrchardCore.Templates/Controllers/TemplateController.cs

```diff
@@ -395,4 +365,32 @@ private IActionResult RedirectToReturnUrlOrIndex(string returnUrl)
             return RedirectToAction(nameof(Index));
         }
     }
+
+    private async Task ValidateModelAsync(TemplateViewModel model, TemplatesDocument templatesDocument = null, string sourceName = null)
+    {
+        if (string.IsNullOrWhiteSpace(model.Name))
+        {
+            ModelState.AddModelError(nameof(TemplateViewModel.Name), S["The name is mandatory."]);
+        }
+        else
+        {
+            templatesDocument ??= model.AdminTemplates
+                ? await _adminTemplatesManager.GetTemplatesDocumentAsync()
+                : await _templatesManager.GetTemplatesDocumentAsync();
+
+            if (!model.Name.Equals(sourceName, StringComparison.OrdinalIgnoreCase) &&
+                templatesDocument.Templates.ContainsKey(model.Name))
+            {
+                ModelState.AddModelError(nameof(TemplateViewModel.Name), S["A template with the same name already exists."]);
+            }
+        }
+
+        if (string.IsNullOrWhiteSpace(model.Content))
+        {
+            var placementsLink = Url.ActionLink("Index", "Admin", new { area = "OrchardCore.Placements" });
```

Comment:
Can we rid off the magic string and use the action, controller names from the helpers